apps/lwip/components/Server/src/eth_interface.c

```diff
@@ -0,0 +1,219 @@
+
+#include <lwip/init.h>
+#include <lwip/netif.h>
+#include <lwip/timeouts.h>
+#include <lwip/udp.h>
+
+#include <netif/etharp.h>
+
+static lwip_iface_t lwip_driver = {0};
+
+static void low_level_init(struct eth_driver *driver, uint8_t *mac, int *mtu)
+{
+    // 1500 is the standard ethernet MTU at the network layer.
+    *mtu = 1500;
+    ethdriver_mac(&mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
+}
+
+extern void *ethdriver_buf;
+
+static void raw_poll(struct eth_driver *driver)
+{
+    int len;
+    int status;
+    status = ethdriver_rx(&len);
+    while (status != -1) {
+        void *buf;
+        void *cookie;
+        buf = (void *)driver->i_cb.allocate_rx_buf(driver->cb_cookie, len, &cookie);
+        if (buf) {
+            // Only proceed if we successfully got a buffer. If not, the packet will simply be dropped.
+            // UDP doesn't mind, and the packet will be resent for TCP due to us not sending an ACK.
+            // This prevents crashing in a DDOS attack or malicious packet that is too large.
+            memcpy(buf, (void *)ethdriver_buf, len);
+            driver->i_cb.rx_complete(driver->cb_cookie, 1, &cookie, (unsigned int *)&len);
+        }
+
+        if (status == 1) {
+            status = ethdriver_rx(&len);
+        } else {
+            /* if status is 0 we already saw the last packet */
+            assert(status == 0);
+            status = -1;
+        }
+    }
+}
+
+static int raw_tx(struct eth_driver *driver, unsigned int num, uintptr_t *phys, unsigned int *len, void *cookie)
+{
+    unsigned int total_len = 0;
+    int i;
+    void *p = (void *)ethdriver_buf;
+    for (i = 0; i < num; i++) {
+        memcpy(p + total_len, (void *)phys[i], len[i]);
+        total_len += len[i];
+    }
+    ethdriver_tx(total_len);
+    return ETHIF_TX_COMPLETE;
+}
+
+static void handle_irq(struct eth_driver *driver, int irq)
+{
+    raw_poll(driver);
+}
+
+void interface_tick() {
+    handle_irq(&lwip_driver.driver, 0);
+}
+
+static struct raw_iface_funcs iface_fns = {
+    .raw_handleIRQ = handle_irq,
+    .print_state = NULL,
+    .low_level_init = low_level_init,
+    .raw_tx = raw_tx,
+    .raw_poll = raw_poll
+};
+
+static int ethdriver_init(struct eth_driver *eth_driver, ps_io_ops_t io_ops, void *config)
+{
+    eth_driver->eth_data = NULL;
+    eth_driver->dma_alignment = 1;
+    eth_driver->i_fn = iface_fns;
+    return 0;
+}
+
+static void *malloc_dma_alloc(void *cookie, size_t size, int align, int cached, ps_mem_flags_t flags)
+{
+    assert(cached);
+    int error;
+    void *ret = malloc(size);
+    if (ret == NULL) {
+        ZF_LOGE("ERR: Failed to allocate %d\n", size);
+        return NULL;
+    }
+    return ret;
+}
+
+static void malloc_dma_free(void *cookie, void *addr, size_t size)
+{
+    free(addr);
+}
+
+static uintptr_t malloc_dma_pin(void *cookie, void *addr, size_t size)
+{
+    return (uintptr_t)addr;
+}
+
+static void malloc_dma_unpin(void *cookie, void *addr, size_t size)
+{
+}
```

Comment:
For empty functions, pleas add a comment in the body that they are empty on purpose. So it's clear the contain was not lonst somewhere during merging or rebasing or. The parameters should also be flagged as unused to silence compiler warnings.